src/cpp/rtps/network/ReceiverResource.cpp

```diff
@@ -45,12 +45,18 @@ ReceiverResource::ReceiverResource(
     }
 
     // Implementation functions are bound to the right transport parameters
-    Cleanup = [&transport, locator]() { transport.CloseInputChannel(locator); };
+    Cleanup = [&transport, locator]()
+            {
+                transport.CloseInputChannel(locator);
+            };
     LocatorMapsToManagedChannel = [&transport, locator](const Locator_t& locatorToCheck) -> bool
-    { return transport.DoInputLocatorsMatch(locator, locatorToCheck); };
+            {
+                return locator.kind == locatorToCheck.kind && transport.DoInputLocatorsMatch(locator, locatorToCheck);
```

Comment:
Only a nitpick. Shared memory transport implementation of `DoInputLocatorsMatch` already checks that the locators `kind` is equal. Maybe it would be better to remove this check from this specific transport implementation so it is not checked twice.

Reply:
This is not used much (only on endpoint matching), so I think it does no harm to perform the check twice.